packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3752,49 +3718,49 @@ export class ContainerRuntime
 		summaryLogger: ITelemetryLoggerExt,
 	): Promise<{ latestSnapshotRefSeq: number; latestSnapshotVersionId: string | undefined }> {
 		const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
-		const { snapshotTree, versionId, latestSnapshotRefSeq } =
-			await this.fetchSnapshotFromStorageAndMaybeClose(
-				summaryLogger,
-				{
-					eventName: "RefreshLatestSummaryFromServerFetch",
-				},
-				readAndParseBlob,
-				null,
-			);
-		const fetchLatestSnapshot: IFetchSnapshotResult = {
-			snapshotTree,
-			snapshotRefSeq: latestSnapshotRefSeq,
-		};
-		const result = await this.summarizerNode.refreshLatestSummary(
-			undefined /* proposalHandle */,
-			latestSnapshotRefSeq,
-			async () => fetchLatestSnapshot,
-			readAndParseBlob,
+		const { versionId, latestSnapshotRefSeq } = await this.fetchSnapshotFromStorage(
 			summaryLogger,
-		);
-
-		// Notify the garbage collector so it can update its latest summary state.
-		await this.garbageCollector.refreshLatestSummary(
-			undefined /* proposalHandle */,
-			result,
+			{
+				eventName: "RefreshLatestSummaryFromServerFetch",
+			},
 			readAndParseBlob,
+			null,
 		);
 
+		await this.closeStaleSummarizer("RefreshLatestSummaryFromServerFetch");
+
 		return { latestSnapshotRefSeq, latestSnapshotVersionId: versionId };
 	}
 
+	private async closeStaleSummarizer(codePath: string): Promise<void> {
+		this.mc.logger.sendTelemetryEvent(
+			{
+				eventName: "ClosingSummarizerOnSummaryStale",
+				codePath,
+				message: "Stopping fetch from storage",
```

Comment:
What is this field for? It's redundant with `eventName`, right?

Reply:
And it will prevent the error's message from being copied over to the event, not that we typically look for that, but JFYI.

Reply:
(I realized this is just a code move, not actually new here, but something minor to consider in your next PRs)